scapy/contrib/isotp.py

```diff
@@ -583,20 +583,6 @@ def can_on_recv(p):
             warning('Provide a basecls ')
         self.basecls = basecls
 
-    def __del__(self):
-        """Close the socket and stop the receiving thread"""
-        self.can_socket.close()
-        self.rx_thread.stop()
-
-    def __enter__(self):
-        """Using the with statement is useful in unit tests executed on an
-        interactive shell, to make sure the receiver thread is stopped"""
-        return self
-
-    def __exit__(self, exc_type, exc_value, traceback):
-        """Close the socket and stop the receiving thread"""
-        self.rx_thread.stop()
-
     def close(self):
         """Close the socket and stop the receiving thread"""
         self.can_socket.close()
```

Comment:
You might want to add `self.rx_thread.stop()` on here so that the feature of the deleted functions is kept ?

Reply:
The next line is already `self.rx_thread.stop()`. Should I swap the lines?